src/stores/ContractsStore.ts

```diff
@@ -1,64 +1,73 @@
-  contractsOwned: Promise<string[]>
-  currentBlock?: number
-  fetchBlockNumber: () => Promise<number>
-  fetchMoreContractsOwned: (accountChange?: boolean) => Promise<void>
-}
+class ContractsStore extends PersistableStore {
+  connectedAccounts: { [account: string]: ContractSynchronizer } = {}
+  currentBlock: undefined | number
 
-const ContractsStore = proxy<ContractsStoreType>({
-  connectedAccounts: {},
-  contractsOwned: Promise.resolve([]),
-  currentBlock: undefined,
+  replacer = (key: string, value: unknown) => {
+    const disallowList = [] as string[] // ['contractsOwned']
+    return disallowList.includes(key) ? undefined : value
+  }
+
+  reviver = (key: string, value: unknown) => {
+    if (key === 'connectedAccounts') {
+      return Object.entries(
+        value as { [account: string]: ContractSynchronizerSchema }
+      ).reduce(
+        (result, [account, state]) => ({
+          ...result,
+          [account]: ContractSynchronizer.fromJSON(state),
+        }),
+        {}
+      )
+    }
+    return value
+  }
 
   fetchBlockNumber() {
     return defaultProvider.getBlockNumber()
-  },
-  async fetchMoreContractsOwned(accountChange) {
-    if (!ContractsStore.currentBlock)
-      ContractsStore.currentBlock = await ContractsStore.fetchBlockNumber()
+  }
+
+  async fetchMoreContractsOwned() {
+    if (!this.currentBlock) this.currentBlock = await this.fetchBlockNumber()
 
     if (!WalletStore.account) {
-      ContractsStore.contractsOwned = Promise.resolve([])
       return
     }
 
-    if (!ContractsStore.connectedAccounts[WalletStore.account])
-      ContractsStore.connectedAccounts[WalletStore.account] =
-        new ContractSynchronizer(WalletStore.account)
-    if (
-      accountChange ||
-      !ContractsStore.connectedAccounts[WalletStore.account] ||
-      !ContractsStore.contractsOwned
-    ) {
-      ContractsStore.contractsOwned = ContractsStore.connectedAccounts[
+    if (!this.connectedAccounts[WalletStore.account]) {
+      this.connectedAccounts[WalletStore.account] = new ContractSynchronizer(
         WalletStore.account
-      ].getOwnedERC721(ContractsStore.currentBlock)
-    } else {
-      const oldContractsOwned = (await ContractsStore.contractsOwned) || []
-      const newContractsOwned = await ContractsStore.connectedAccounts[
-        WalletStore.account
-      ].getOwnedERC721(ContractsStore.currentBlock)
-      ContractsStore.contractsOwned = Promise.resolve(
-        Array.from(new Set([...oldContractsOwned, ...newContractsOwned]))
       )
     }
-  },
-})
+
+    await this.connectedAccounts[WalletStore.account].getOwnedERC721(
+      this.currentBlock
+    )
+  }
+}
+
+const contractsStore = proxyWithComputed(new ContractsStore(), {
+  contractsOwned: (state) =>
+    walletStore.account && state.connectedAccounts[walletStore.account]
+      ? state.connectedAccounts[walletStore.account].contractsOwned
+      : [],
+}).makePersistent(true)
 
 subscribeKey(WalletStore, 'account', () =>
```

Comment:
Maybe we can pass `subscribeKey(WalletStore, 'account', contractStore.fetchMoreContractsOwned)`?

Reply:
Here we can’t do this, the context is lost